File: src/protocol/prompts/chooseTargetCard.ts
import type { TargetingIntent } from "@/types/promptType";

export type Type = "chooseTargetCard";
export type Input = {
  type: Type;
  validCardIds: string[];
  hostile?: boolean;
  intent: TargetingIntent;
  minTargets: number;
  maxTargets: number;
  chosenTargets: number;
};
export type Output = { type: "targetCard"; cardId: string | null };

File: forge-engine/crates/forge-protocol/src/prompts/choose_target_card.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

use crate::values::TargetingIntent;

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export, export_to = "prompts/chooseTargetCard.ts")]
pub struct ChooseTargetCardInput {
    pub valid_card_ids: Vec<String>,
    #[serde(default)]
    pub hostile: bool,
    pub intent: TargetingIntent,
    pub min_targets: i32,
    pub max_targets: i32,
    pub chosen_targets: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[serde(
    tag = "type",
    rename_all = "camelCase",
    rename_all_fields = "camelCase"
)]
#[ts(export, export_to = "prompts/chooseTargetCard.ts")]
pub enum ChooseTargetCardOutput {
    TargetCard { card_id: Option<String> },
}
